src/main/java/spotify/factories/AbstractPlayableObjectDeserializer.java

```diff
@@ -7,13 +7,46 @@
 
 import java.lang.reflect.Type;
 
+/**
+ * This is a AbstractPlayableObjectDeserializer class to implement JsonDeserializer.
+ *
+ * @see spotify.models.generic.AbstractPlayableObject
+ * @author Jiankai Zheng (jk.zheng@hotmail.com)
+ * @since // check with Jiankai
+ */
+
 class AbstractPlayableObjectDeserializer implements JsonDeserializer<AbstractPlayableObject> {
+
+    /**
+     * Custom deserialize method to deserialize json, type and jsonDeserializationContext
+     * @param json represents a json element
+     * @param type represents various types
+     * @param jsonDeserializationContext
+     * @return childClass
```

Comment:
Not descriptive enough, child class of what?

Reply:
Would this be better?

childClass that deserialize JSON string either "track" or "episode" into Java objects

Reply:
No it should tell that it's returning a object which is a subclass of AbstractPlayableObject

Reply:
Thanks. I will update the documentation.